design/ceph/object/ceph-rgw-hpa-handle-reconcile.md

```diff
@@ -0,0 +1,43 @@
+---
+title: handle object store reconcilation for HPA
+target-version: release-1.10 (preferably backported to 1.9)
+---
+
+# Feature Name
+Handle object store reconcilation for Horizontal Pod Autoscaler(HPA)
+
+
+## Summary
+If Horizontal Pod Scalers is configured scaled up replica count for ceph object store, then during reconcilation Rook Operator will reset the value for replica and suddenly HPA set it back. In my testing for 5 replica count, during reconcilaton it went to 1 for 5-10s and HPA recreated replicas 4 after that. To avoid this scenario reconcilation check whether HPA is configured or not and set the replica value according to that.
+
+### Goals
+Avoid resetting on replica count by Rook Operator for RGW deployment if HPA configured.
+
+
+### Non-Goals
+Make it option in `CephObjectStoreSpec` than default.
+
+## Proposal details
+The HPA need to have label referring to `cephobjectstore`, during the reconcilation fetch the current replication count set by HPA and update it deployment for RGW.
+
+## Drawbacks
+The user need to add label for HPA.
+
+## Alternatives
+Instead of adding label for the HPA, Rook can fetch all the HPAs in the check the `deployment` from `scaleTargetRef`. But it may increase the time for reconcilation if there a lot of HPAs in the namespace. Or label can be added on `cephobjectstore` about the HPA.
+Configure HPA directly on the `cephobjectstore` CR, need to define `scale` subresource for that.
```

Comment:
i think this might be better. this would make it clear which number we need to look at.
this would allow us to manually override (increase or decrease) the HPA value

Reply:
@yuvalif : please refer [Line42](https://github.com/rook/rook/pull/10316/files#diff-b69f2b2e58d760d62f0452e00e9f026523e5036c5fd86c1f42851bd6ef12bc57R42)

Reply:
Please see [comment](https://github.com/rook/rook/pull/10316#discussion_r882892181)